src/components/Form/Widgets/FileWidget.tsx

```diff
@@ -0,0 +1,218 @@
+				);
+				setErrors(errs);
+				return onChange();
+			}
+
+			acceptedFiles.forEach((file) => {
+				const reader = new FileReader();
+				reader.onerror = () => {
+					const err = [`Failed to upload ${file.name}`];
+					onChange();
+					setErrors(err);
+				};
+				reader.onloadstart = () => setLoadingPercentage(0);
+				reader.onprogress = (event) => {
+					setLoadingPercentage(Math.floor((100 * event.loaded) / event.total));
+				};
+				reader.onload = () => {
+					if (file != null && reader.result != null) {
+						setLoadingPercentage(undefined);
+						onChange({
+							dataUrl: reader.result as string,
+							uploadedFile: file,
+						});
+					}
+				};
+
+				reader.readAsDataURL(file);
+			});
+		}) as NonNullable<DropzoneOptions['onDrop']>,
+		[onChange, setLoadingPercentage, setErrors],
+	);
+
+	const { getRootProps, getInputProps, isDragActive } = useDropzone({
+		onDrop,
+		maxSize,
+		accept,
+		multiple: false,
+		noDrag: false,
+		disabled: loadingPercentage !== undefined,
+	});
+
+	return (
+		<>
+			{schema.title && <InputLabel>{schema.title}</InputLabel>}
+			<Box
+				sx={{
+					border: (theme) =>
+						`dashed 2px ${
+							errors.length !== 0
+								? '#F19B9E'
+								: isDragActive
+								? theme.palette.primary.main
+								: 'rgba(0, 0, 0, 0.23)'
+						}`,
+					borderRadius: '2px',
+					display: 'flex',
+					flexDirection: 'column',
+					alignItems: 'center',
+					justifyContent: 'center',
+				}}
+				style={{ minHeight: '150px', cursor: 'pointer' }}
+				flexDirection="column"
+				alignItems="center"
+				justifyContent="center"
+				{...getRootProps()}
+			>
+				<Input {...getInputProps()} />
+				{loadingPercentage !== undefined && (
+					<Box
+						sx={{
+							position: 'absolute',
+							top: 0,
+							left: 0,
+							width: '100%',
+							height: '100%',
+							display: 'flex',
+							alignItems: 'center',
+							justifyContent: 'center',
+							background: 'rgba(255, 255, 255, 0.7)',
+						}}
+					>
+						<CircularProgress
+							id="loading-progress"
+							variant="determinate"
+							value={loadingPercentage}
+						/>
+					</Box>
+				)}
+
+				<Box
+					aria-busy={loadingPercentage !== undefined}
+					aria-describedby="loading-progress"
+					display="flex"
+					alignItems="center"
+					justifyContent="center"
+				>
+					{value ? (
+						isImage(value) ? (
+							<Avatar
+								sx={{
+									borderRadius: 0,
+									height: '100px',
+									width: '100px',
+									backgroundSize: 'contain',
+								}}
+								src={dataUrl(value)}
+							/>
+						) : (
+							<FilePresentIcon
+								sx={{
+									fontSize: 64,
+									color: (theme) => theme.palette.grey[500],
+								}}
+							/>
+						)
+					) : (
+						<UploadFileIcon
+							sx={{ fontSize: 64, color: (theme) => theme.palette.grey[500] }}
+						/>
+					)}
+					<Typography>Drag and drop or browse files</Typography>
```

Comment:
```suggestion
					<Typography ml={2}>Drag and drop or browse files</Typography>
```

Oops, forgot to suggest this in the last suggestion, my bad